modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -320,11 +320,18 @@ bool GifDecoder::lzwDecode() {
     const int lzwMaxSize = (1 << 12); // 4096 is the maximum size of the LZW table (12 bits)
     int lzwCodeSize = lzwMinCodeSize + 1;
     CV_Assert(lzwCodeSize > 2 && lzwCodeSize <= 12);
-    int clearCode = 1 << lzwMinCodeSize;
-    int exitCode = clearCode + 1;
-    std::vector<lzwNodeD> lzwExtraTable(lzwMaxSize + 1);
-    int colorTableSize = clearCode;
-    int lzwTableSize = exitCode;
+    const int clearCode = 1 << lzwMinCodeSize;
+    const int exitCode = clearCode + 1;
+    std::vector<lzwNodeD> lzwExtraTable;
```

Comment:
Why do you remove `lzwMaxSize+1` here? If I understand correctly the issue is related to exitCode state. It clears the vector and it's used after .clear() call. the vector may be used becore clearCode and it's not allocated properly.

Reply:
I removed it in case blockLen is 0 in the first place (not sure if that can happen), this way we have one allocation less. `clear()` is always called before using `lzwExtraTable`, at the beginning of the while loop.